march_hardware/src/Joint.cpp

```diff
@@ -94,6 +96,27 @@ float Joint::getTemperature()
   return this->temperatureGES.getTemperature();
 }
 
+std::vector<std::string> Joint::getIMotionCubeState()
+{
+    // Return a vector of strings:
+    // First the literal bits of the Status Word, Detailed Error and Motion Error
+    // Then the parsed interpretation of these bits
+    std::vector<std::string> return_strings;
+
+    std::bitset<16> statusWordBits = this->iMotionCube.getStatusWord();
+    return_strings.push_back(statusWordBits.to_string());
+    std::bitset<16> detailedErrorBits = this->iMotionCube.getDetailedError();
+    return_strings.push_back(detailedErrorBits.to_string());
+    std::bitset<16> motionErrorBits = this->iMotionCube.getMotionError();
+    return_strings.push_back(motionErrorBits.to_string());
+
+    return_strings.push_back(this->iMotionCube.getState(this->iMotionCube.getStatusWord()));
+    return_strings.push_back(this->iMotionCube.parseDetailedError(this->iMotionCube.getDetailedError()));
+    return_strings.push_back(this->iMotionCube.parseMotionError(this->iMotionCube.getMotionError()));
+
+    return return_strings;
```

Comment:
Using a vector to return multiple values is discourages, consider creating your own object instead